### PR TITLE
Make ownCloud work again in php 7.0.6

### PR DESCRIPTION
**Pull request by rullzer**
See https://bugs.php.net/bug.php?id=72117

CC: @nickvergessen @PVince81 @LukasReschke @DeepDiver1975

### CODE REVIEW

File: lib/private/AppFramework/Http/Request.php [hunk 1]
```diff
@@ -271,6 +271,9 @@ public function __get($name) {
 	 * @return bool
 	 */
 	public function __isset($name) {
+		if (in_array($name, $this->allowedKeys, true)) {
```
Comment (nickvergessen) [inline lib/private/AppFramework/Http/Request.php]: use hasMember?

Comment (rullzer) [inline lib/private/AppFramework/Http/Request.php]: how?

**Comment by oparoz** [comment]
Is Jenkins using PHP 7.0.6?

**Comment by oparoz** [comment]
Because Travis isn't:
https://travis-ci.org/owncloud/core/jobs/126342764#L183

**Comment by RealRancor** [comment]
> Is Jenkins using PHP 7.0.6?

PHP 7.0.6 is not even officially announced:
https://secure.php.net/ChangeLog-7.php
https://secure.php.net/archive/2016.php

**Comment by oparoz** [comment]
@RealRancor - Of course, but what I'm saying is that we can't be almost absolutely certain that this is the only required change unless the branch uses PHP 7.0.6 for tests.
A newer PHP can be compiled on the CI boxes or we can wait for Travis to catch up, which will take months as they're still using 7.0.4.

**Comment by rullzer** [comment]
I ran the unit test locally with php 7.0.6 (which is tagged alrady) and now they pass.

**Comment by rullzer** [comment]
And at least we know it is not breaking older isntances.

**Comment by maki-chan** [comment]
As an Arch Linux user, I updated to PHP 7.0.6, too, as it was available on the package mirrors during the update I made today. This patch does indeed work and I can use ownCloud on my server again 👍

**Comment by HLFH** [comment]
I'm also an Arch Linux user and I updated to PHP 7.0.6. This PR fixes the issue. 👍

**Comment by rullzer** [comment]
@karlitschek we should backport this. Else users on stable 8.2 and 9.0 might run into trouble if they run bleeding edge php

**Comment by rullzer** [comment]
Stable 9: #24343 
Stable8.2: #24344

**Comment by karlitschek** [comment]
backport makes sense 👍

**Comment by jirkaehm** [comment]
Just spent about two hours debugging why owncloud stopped working. Find out that my webhost upgraded to PHP 7.0.6. Fix mentioned above did the trick.

**Comment by produnis** [comment]
this patch worked for me, too!
THX A LOT!!!!

**Comment by nonamedotc** [comment]
Thank you for the patch. I was able to fix up my installation as well!

:+1:

File: lib/private/AppFramework/Http/Request.php (hunk 1 above)
Comment (thomasvnl) [inline lib/private/AppFramework/Http/Request.php]: This worked for me. Pulled a PHP 7 update from Ubuntu's official update stream and my ownCloud broke down. Thanks to all who came up with a patch!

Comment (RealRancor) [inline lib/private/AppFramework/Http/Request.php]: @thomasvnl You don't need this patch. Just use the current oC version 9.0.4 which already includes this.

Comment (thomasvnl) [inline lib/private/AppFramework/Http/Request.php]: @RealRancor I understand, but yesterday I didn't have time to update my ownCloud from 9.0.0 to 9.0.4. I also noticed the fix isn't mentioned in the changelog, or am I mistaken? I've searched through the changelog first before going this route. 

Thanks for notifying me about this fix being included in 9.0.4.

Comment (PVince81) [inline lib/private/AppFramework/Http/Request.php]: From what I see the fix was already in 9.0.2 https://github.com/owncloud/core/pull/24343

Comment (RealRancor) [inline lib/private/AppFramework/Http/Request.php]: Yes, and also listed in the (quite large) ChangeLog of 9.0.2:

> [Stable 9] Make ownCloud work again in php 7.0.6 - #24343

https://owncloud.org/changelog/

Comment (thomasvnl) [inline lib/private/AppFramework/Http/Request.php]: @PVince81 @RealRancor totally missed it! Thanks guys.

**Pull request merged**